sklearn/utils/_array_api.py

```diff
@@ -1001,3 +1019,9 @@ def _count_nonzero(X, xp, device, axis=None, sample_weight=None):
 
     zero_scalar = xp.asarray(0, device=device, dtype=weights.dtype)
     return xp.sum(xp.where(X != 0, weights, zero_scalar), axis=axis)
+
+
+def _xp_method_has_out(xp_method):
+    out_param = "out=None"
+    method_doc = xp_method.__doc__
+    return method_doc and out_param in method_doc
```

Comment:
I think this kind of docstring inspection too brittle. Either do the try/except TypeError pattern suggested in the other review comment or explicitly test for `if _is_numpy_namespace(xp)`.